### PR TITLE
add warning for the removal of extend in v4

### PR DESCRIPTION
**Pull request by probablyup**

### CODE REVIEW

**Comment by probablyup** [comment]
That sounds like a lot of backbending. The React 16 upgrade was pretty easy and it's been out long enough that I'm not concerned.

**Comment by mxstbr** [comment]
Sure, then let's just warn about extend!

File: src/models/StyledNativeComponent.js [hunk 1]
```diff
@@ -17,6 +18,16 @@ type State = {
   generatedStyles: any,
 }
 
+let warnExtendDeprecated
+if (process.env.NODE_ENV !== 'production') {
+  warnExtendDeprecated = once(() => {
+    // eslint-disable-next-line no-console
+    console.error(
+      'Warning: The "extend" API has been removed in the upcoming v4.0 release. Use styled(StyledComponent) instead. You can find more information here: https://github.com/styled-components/styled-components/issues/1546'
```
Comment (mxstbr) [inline src/models/StyledNativeComponent.js]: Maybe change the wording to `The "extend" API will be removed... ...due to issues with server-side rendering.`?

Comment (probablyup) [reply src/models/StyledNativeComponent.js]: There were other reasons too. Linking to the issue isn't enough?

Comment (mxstbr) [reply src/models/StyledNativeComponent.js]: Oh yeah I didn't see that, but we should still say "will be"?

Comment (probablyup) [reply src/models/StyledNativeComponent.js]: Sure

**Comment by probablyup** [comment]
@mxstbr done

**Comment by mxstbr** [comment]
Yesss

**Pull request merged**